File: original/94. Binary Tree Inorder Traversal/94.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        arr = []
        if root == None:
            return arr
        
        else:
            if root.left:
                arr += self.inorderTraversal(root.left)    
            
            arr.append(root.val)
            
            if root.right:
                arr += self.inorderTraversal(root.right)
            
            return arr
            

File: sprint/binaryTree/recursionTraversal/94.cpp
#include <vector>
using namespace std;

struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
};

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        traversal(root, res);   
        return res;
    }

    void traversal(TreeNode* root, vector<int>& res) {
        if (root == nullptr) {
            return;
        }
        traversal(root->left, res);
        res.push_back(root->val);
        traversal(root->right, res);
    }
};

int main(void) {
    TreeNode* t = new TreeNode();
}